---
title: "HPV-induced host epigenetic reprogramming is lost upon progression to high-grade cervical intraepithelial neoplasia"
author: "Main Figures"
date: ""
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
header-includes:
  \usepackage{fontspec}
  \usepackage[utf8]{inputenc}
  \usepackage{pdfpages}
  \usepackage{graphicx}
  \setmainfont{Arial}
---

```{r, echo=FALSE, message = FALSE, warning = FALSE}
library(knitr)
library(ggplot2)
library(patchwork)
library(dplyr)
library(tidyverse)
library(ggbeeswarm)
library(gghalves)
library(pROC)
library(ggpubr)
#library(kableExtra)
library(gtsummary)
library(gt)
theme_gtsummary_compact()

knitr::opts_chunk$set(echo=FALSE, message = FALSE, warning = FALSE)

theme_set(theme_minimal())

```


Chiara Herzog\*, Charlotte D. Vavourakis\*, James E. Barrett, Gerlinde Karbon, Andreas Villunger, Jiangrong Wang, Karin Sundstr√∂m, Joakim Dillner, Martin Widschwendter


\* contributed equally

\tableofcontents

\pagebreak

```{r, warning = FALSE, message=FALSE, fig.width=15, fig.height=8}
# # -------- Cell composition in training (Fig 1B)
# dat_cin <- readRDS("../0-data/pheno_P1.Rds")
# 
# cin.colours2 <- c("#00A08799","#3C548899", "#7E614899", "#F39B7F99", "#DC000099")
# 
# dat <- dat_cin %>%
#   pivot_longer(grep("hepidish", colnames(dat_cin)),
#                names_to = "celltype",
#                values_to = "value")
# 
# dat$set <- dat$split
# 
# celltypes <- dat %>%
#   filter(set == "training set") %>%
#   mutate(type = case_when(type == "Control_NEG" ~ "HPV-",
#                           type == "Control_POS" ~ "HPV+"),
#          celltype = case_when(celltype == "hepidish_NK" ~ "NK cells",
#                               celltype == "hepidish_Neutro" ~ "Neutrophils",
#                               celltype == "hepidish_Mono" ~ "Monocytes",
#                               celltype == "hepidish_Fib" ~ "Fibroblasts",
#                               celltype == "hepidish_Epi" ~ "Epithelial cells",
#                               celltype == "hepidish_Eosino" ~ "Eosinophils",
#                               celltype == "hepidish_CD8T" ~ "CD8+ T cells",
#                               celltype == "hepidish_CD4T" ~ "CD4+ T cells",
#                               celltype == "hepidish_B" ~ "B cells")) %>%
#   mutate(celltype = factor(celltype, levels = c("Eosinophils",
#                                                 "CD8+ T cells",
#                                                 "CD4+ T cells",
#                                                 "B cells",
#                                                 "NK cells",
#                                                 "Monocytes",
#                                                 "Fibroblasts",
#                                                 "Neutrophils",
#                                                 "Epithelial cells"))) %>%
#   ggplot(aes(x = celltype,
#              y = value,
#              fill = type)) +
#   geom_boxplot() +
#   coord_flip() +
#   stat_compare_means(method = "wilcox.test",
#                      aes(label = paste0("p=", ..p.format..)),
#                      label.y = 0.9,
#                      colour = "gray40",
#                      size = 2.75,) +
#   theme_classic() +
#   theme(legend.position = "top",
#         legend.title = element_blank(),
#         axis.ticks = element_blank()) +
#   scale_fill_manual(values = cin.colours2[c(1,2)],
#                     aesthetics = "fill") +
#   xlab("") +
#   ylab("Proportion")
# 
# ggsave(celltypes, filename = "../0-data/figures/fig1b.pdf",
#        width = 3.1, height = 3.5, dpi = 300)
# 
# #-------- DELTA BETA histogram plot (Fig 1C)
# dat <- readRDS("../0-data/delta-beta.Rds")
# 
# cin.colours2 <- c("#00A08799","#3C548899", "#7E614899", "#F39B7F99", "#DC000099")
# 
# dbplot <- ggplot(dat,aes(x=db))+
#   geom_histogram(data=subset(dat,Tissue=='Epithelial'),
#                  aes(y=..count../sum(..count..),
#                      fill=Tissue),
#                  alpha=0.7,
#                  binwidth=0.001)+
#   geom_histogram(data=subset(dat,Tissue=='Immune'),
#                  aes(y=..count../sum(..count..),
#                      fill=Tissue),
#                  alpha=0.6,
#                  binwidth=0.001)+
#   scale_fill_manual(name="Cell type",
#                     values=cin.colours2[c(4,2)],
#                     labels=c("Epithelial","Immune")) +
#   xlim(-0.05,0.05) +
#   theme_minimal() +
#   xlab("Delta-beta estimate") +
#   ylab("Frequency") +
#   theme(panel.grid = element_blank(),
#         legend.position = "top")
# 
# ggsave(dbplot, filename = "../0-data/figures/fig1c.pdf",
#        width = 3.5, height = 3.5, dpi = 300)
# 
# 
# 
# #-------- Training - N (Figure 1D)
# n_tr <- readRDS("../0-data/n_training.Rds")
# trplot <- n_tr %>%
#   ggplot(aes(x = n_seq,
#              y = oob_auc)) +
#   geom_line(size = 1,
#             colour = cin.colours2[1]) +
#   geom_point(aes(x = 500,
#                  y = 0.8384676),
#              size = 2,
#              shape = 4) +
#   theme(panel.grid = element_blank(),
#         panel.background = element_blank()) +
#   coord_cartesian(xlim = c(0,5000)) +
#   xlab("Number of top differentially methylated CpGs\nused as input") +
#   ylab("Out-of-bag AUC estimate")
# ggsave(trplot, filename = "../0-data/figures/fig1d.pdf",
#        width = 3.5, height = 3.5, dpi = 300)
# 
# # WID-HPV v immune cell proportion/age (Figure 1E/F)
# dat_cin <- readRDS("../0-data/pheno_P1.Rds")
# 
# # adjust WID_HPV for age and ic
# fit <- lm(WID_HPV ~ age + ic, data = dat_cin[dat_cin$type=="Control_NEG" & dat_cin$split == "hold out test set",])
# dat_cin$WID_HPV_adj <- dat_cin$WID_HPV - predict(fit, newdata = dat_cin)
# 
# # arrange factor "type" (sample groups) for uniform plotting
# dat_cin$type <- factor(dat_cin$type, levels = c("Control_NEG", "Control_POS", "CIN1", "CIN2", "CIN3+"))
# 
# type.labs1 <- c("HPV-/cyt-", "HPV+/cyt-")
# names(type.labs1) <- c("Control_NEG", "Control_POS")
# cin.colours1 <- c("#00A08799","#3C548899")
# names(cin.colours1) <- c("Control_NEG", "Control_POS")
# 
# icplot <- dat_cin %>% 
#   filter(split == "hold out test set" & type %in% c("Control_NEG","Control_POS")) %>%
#   droplevels() %>%
#   ggplot(aes(x = ic,
#              y = WID_HPV,
#              colour = type)) +
#   geom_point() +
#   geom_smooth(method = "lm",
#               se = FALSE) +
#   xlab("sample immune cell proportion") +
#   ylab("WID-HPV") +
#   theme(aspect.ratio = 1) +
#   scale_colour_manual(name="",labels=type.labs1,values=cin.colours1) 
# ggsave(icplot, filename = "../0-data/figures/fig1e.pdf",
#        width = 4.75, height = 3.5, dpi = 300)
# 
# ageplot  <- dat_cin %>% 
#   filter(split == "hold out test set" & type %in% c("Control_NEG","Control_POS")) %>%
#   droplevels() %>%
#   ggplot(aes(x = age,
#              y = WID_HPV,
#              colour = type)) +
#   geom_point() +
#   geom_smooth(method = "lm",
#               se = FALSE) +
#   xlab("sample donor age") +
#   ylab("WID-HPV") +
#   theme(aspect.ratio = 1) +
#   scale_colour_manual(name="",labels=type.labs1,values=cin.colours1) 
# ggsave(ageplot, filename = "../0-data/figures/fig1f.pdf",
#        width = 4.75, height = 3.5, dpi = 300)
# 
# # ROC-curve (Figure 1G)
# source("../1-source-codes/plot_roc_slim.R")
# dat_diseasefree <- dat_cin %>%
#   filter(split == "hold out test set" & type %in% c("Control_NEG", "Control_POS")) %>%
#   droplevels()
# fit <- lm(WID_HPV ~ age + ic, data = dat_diseasefree[dat_diseasefree$type=="Control_NEG",])
# dat_diseasefree$WID_HPV_adj <- dat_diseasefree$WID_HPV - predict(fit, newdata = dat_diseasefree)
# auc <- plot_roc(dat_diseasefree$type,
#                 dat_diseasefree$WID_HPV_adj)
# ggsave(auc, filename = "../0-data/figures/fig1g.pdf",
#        width = 3.5, height = 3.5, dpi = 300)
# 
```

\includegraphics[trim=0 0cm 0 0, clip]{../0-data/figures/Fig_1.pdf}

\subsection{Figure 1. Study overview and WID-HPV training and evaluation.} \textbf{A} Study overview. Early HPV-associated host epigenome changes in the Cervix uteri were evaluated in a sample set of (cervical) disease-free, cytology-negative HPV+ or HPV- samples. Samples were split 70-30 into a training and test set and differential methylated probes determined. A HPV signature, called WID-HPV (Women's cancer risk-human papillomavirus) was derived using penalized regression models. Performance was validated in the hold-out test set. The WID-HPV was then further evaluated in association with cervical intraepithelial neoplasia progression (CIN1-3+), and its relation with cell-intrinsic factors that may influence defence mechanisms was investigated via the association with replicative age and apoptosis signatures. \textbf{B} Sample cell type composition of HPV+ and HPV- cytology negative samples. \textbf{C} Histogram of differential methylation in epithelial and immune cells, the major cell subtypes in cervical samples. \textbf{D} Out-of-bag area under the curve (AUC) estimate for the receiver operating characteristic during training of the linear WID-HPV index using penalised regression. Providing the top 500 differentially methylated epithelial CpGs as input for an L1 norm classifier (Lasso) resulted in the highest out-of-bag AUC. \textbf{E} WID-HPV versus sample immune cell proportion in the test set. \textbf{F} WID-HPV versus sample donor age in the test set. \textbf{G} Receiver operating characteristic and area under the curve (AUC) for the WID-HPV (adjusted for sample donor age and immune cell proportion, using HPV- cyt- samples) in the test set.

\pagebreak

```{r, warning = FALSE, message=FALSE, fig.width=12,fig.height=8}
## load and arrange CIN data
dat_cin <- readRDS("../0-data/pheno_P1.Rds")

# adjust WID_HPV for age and ic
fit <- lm(WID_HPV ~ age + ic, data = dat_cin[dat_cin$type=="Control_NEG" & dat_cin$split == "hold out test set",]) # adjusting using controls only (from hold out test set)
dat_cin$WID_HPV_adj <- dat_cin$WID_HPV - predict(fit, newdata = dat_cin) # Obtaining adjusted value (residuals) after accounting for differences in age and IC

# arrange factor "type" (sample groups) for uniform plotting
dat_cin$type <- factor(dat_cin$type, levels = c("Control_NEG", "Control_POS", "CIN1", "CIN2", "CIN3+"))
type.labs <- c("HPV-ve/cyt-ve", "HPV+ve/cyt-ve", "CIN1", "CIN2", "CIN3+")
names(type.labs) <- c("Control_NEG", "Control_POS", "CIN1", "CIN2", "CIN3+")
type.labs.3 <- c("HPV+/cyt-", "CIN1", "CIN2", "CIN3+") #HPV-v/cyt-ve removed
names(type.labs.3) <- c("Control_POS", "CIN1", "CIN2", "CIN3+")
cin.colours <- c("#00A087FF", "#3C5488FF", "#7E6148FF", "#F39B7FFF", "#DC0000FF")
cin.colours2 <- c("#00A08799","#3C548899", "#7E614899", "#F39B7F99", "#DC000099")
cin.colours3 <- c("#3C548899", "#7E614899", "#F39B7F99", "#DC000099")
names(cin.colours) <- c("Control_NEG", "Control_POS", "CIN1", "CIN2", "CIN3+")
names(cin.colours2) <- c("Control_NEG", "Control_POS", "CIN1", "CIN2", "CIN3+")
names(cin.colours3) <- c("Control_POS", "CIN1", "CIN2", "CIN3+")

# subplot violins ctrls -> CIN3+ (Figure 2A)
# type.labs[names(type.labs) == "Control_NEG"] <- "HPV-"
# type.labs[names(type.labs) == "Control_POS"] <- "HPV+"

tmp <- dat_cin[dat_cin$split != "training set",]
# HPV+/HPV-
ref <- tmp$type == "Control_NEG"
comp <- tmp$type == "Control_POS"
hpvpos <- signif(p.adjust(wilcox.test(tmp[ref,]$WID_HPV_adj,tmp[comp,]$WID_HPV_adj)$p.value,
                          n = 5,
                          method = "bonferroni"), 2)
 
comp <- tmp$type == "CIN1"
cin1 <- signif(p.adjust(wilcox.test(tmp[ref,]$WID_HPV_adj,tmp[comp,]$WID_HPV_adj)$p.value,
                          n = 5,
                          method = "bonferroni"), 2)

comp <- tmp$type == "CIN2"
cin2 <- signif(p.adjust(wilcox.test(tmp[ref,]$WID_HPV_adj,tmp[comp,]$WID_HPV_adj)$p.value,
                         n = 5,
                         method = "bonferroni"), 2)

comp <- tmp$type == "CIN3+"
cin3 <- p.adjust(wilcox.test(tmp[ref,]$WID_HPV_adj,tmp[comp,]$WID_HPV_adj)$p.value,
                  n = 5,
                  method = "bonferroni")
 
# All + v CIN3+ is significant
ref <- tmp$type %in% c("Control_POS", "CIN1", "CIN2")
comp <- tmp$type == "CIN3+"
cin3_all <- signif(p.adjust(wilcox.test(tmp[ref,]$WID_HPV_adj,
                    tmp[comp,]$WID_HPV_adj)$p.value,
                     n = 5,
                     method = "bonferroni"),2)
cin3_all <- sub("e","%.% 10^",cin3_all)#fix superscript scientific notation
 
p1 <- tmp %>%
   filter(split != "training set") %>%
   ggplot(aes(x = type,
              y = WID_HPV_adj,
              fill=type)) +
   geom_half_boxplot(width=0.2)+
   geom_half_violin() +
   geom_half_point(transformation = position_jitter(),size=0.3,aes(color=type)) +
   ylab("WID-HPV") +
   xlab("") +
   scale_fill_manual(name="",labels=type.labs,values=cin.colours2) +
   scale_color_manual(name="",labels=type.labs,values=cin.colours2) +
   geom_hline(yintercept=0,linetype="dashed",color = "black") +
   theme_classic() +
   theme(axis.text.x=element_blank(),
         axis.ticks.x=element_blank()) +
   annotate("segment",
            x = c(2, 4, 2, 3, 3, 5),
            xend = c(2, 4, 4, 3, 5, 5),
            y = c(3.5, 3.5, 3.7, 3.7, 4, 4),
            yend = c(3.7, 3.7, 3.7, 4, 4, 3.2),
            size = 0.5) +
   annotate("text",
            x = 4,
            y = 4.2,
            label = paste0("p == ", cin3_all),
            size = 3.1,
            colour = "gray40",
            parse=TRUE)

# remove samples used for training and HPV-ve/cyt-ve group
dat <- dat_cin %>% filter(split != "train") %>% filter(type != "Control_NEG") %>% droplevels()

# frequency subtypes in each group (Figure 2B)
# gsub oncogenic HPV for oncogenic HPV
dat$HPV_genotype <- gsub("HR", "oncogenic HPV", dat$HPV_genotype)
dat$HPV_genotype <- factor(dat$HPV_genotype, levels=c("HPV 16", "HPV 16 and other oncogenic HPV", "HPV 16 and 18", "HPV 16, 18 and other oncogenic HPV", "HPV 18", "HPV 18 and other oncogenic HPV", "other oncogenic HPV", "no HPV detected")) #arrange levels

# genotype.colors <- c("#D9565C", "#EA7B7C", "#EF9798", "#CFAAAB", "#39B4AE", "#109DB7", "#0C6EA5", "#172869")
genotype.colors <- c("#B25D91", "#C37AAA", "#DAA2C9", "#D0C4DE", "#39B8B6", "#109DB7", "#0C6EA5", "#172869")

p2 <- dat %>%
  ggplot(aes(x = type, fill = HPV_genotype)) + 
  geom_bar(position = "fill") +
  ylab("Proportion") +
  xlab("") +
  labs(fill="") +
  scale_x_discrete(labels = type.labs) +
  scale_fill_manual(values=genotype.colors) +
  theme_minimal()

#dat_tab <- table(dat$HPV_genotype, dat$type)#contingecy table
#dat_tab <- prop.table(dat_tab, margin=2) #relative frequencies, with sum of each type equal to 1 
#barplot(dat_tab)

# HPV16+HPV18 in relation to HPV (Figure 2C)
dat2 <- dat %>% 
  filter(!HPV_genotype %in% c("no HPV detected", NA)) %>% #remove samples where no HPV was detected
  mutate(HPV_genotype2 = case_when(HPV_genotype == "other oncogenic HPV" ~ "other oncogenic HPV", TRUE ~ "HPV16 and/or HPV18")) # rearrange genotypes

genotype2.colors <- c("#C37AAA","#0C6EA5")

p3 <- dat2 %>%
  ggplot(aes(x = type,
             y = WID_HPV_adj, fill=HPV_genotype2)) +
  geom_half_boxplot()+
  geom_half_point(transformation = position_jitter(),
                  size=0.3,aes(color=HPV_genotype2)) +
  ylab("WID-HPV") +
  xlab("") +
  labs(fill="", color="") +
  scale_x_discrete(labels = type.labs) +
  scale_fill_manual(values=genotype2.colors) +
  scale_color_manual(values=genotype2.colors) +
  geom_hline(yintercept=0, linetype="dashed", color = "black")

# correlation WID-HPV and pcgtage (Figure 2D)
dat_cin_notrain <- dat_cin %>% filter(split != "training set") %>% droplevels()

#first save correlation coefficients and p-values so can format them as wanted
dat3 <- dat_cin_notrain %>%
  filter(type %in% c("Control_POS","CIN1","CIN2","CIN3+")) %>%
  droplevels()

tauCor <- round(cor(dat3$pcgtage,dat3$WID_HPV_adj, method="kendall"),2)
tauPval <- sub("e","%.% 10^",signif(cor.test(dat3$pcgtage,dat3$WID_HPV_adj, method="kendall")$p.value,2))
rhoCor <- round(cor(dat3$pcgtage,dat3$WID_HPV_adj, method="spearman"),2)
rhoPval <- sub("e","%.% 10^",signif(cor.test(dat3$pcgtage,dat3$WID_HPV_adj, method="spearman")$p.value,2))

p4 <- dat3 %>%
  ggplot(aes(x = pcgtage,
             y = WID_HPV_adj)) +
  geom_point(aes(color=type), size=0.6) +
  scale_color_manual(name="",labels=type.labs.3,values=cin.colours3) +
  scale_x_continuous(trans="log10") +
  theme_classic() +
  theme(panel.grid = element_blank()) +
  labs(y = "WID-HPV", x = "log(pcgtAge)", colour="") +
  annotate("text",
           x = 0.2,
           y = 2.5,
           label = paste0(expression(tau), "==", tauCor, "~(", "p == ", tauPval, ")"),
           size = 3.1,
           colour = "gray40",
           parse = TRUE) +
  annotate("text",
           x = 0.2,
           y = 2,
           label = paste0(expression(rho), "==", rhoCor, "~(", "p == ", rhoPval, ")"),
           size = 3.1,
           colour = "gray40",
           parse = TRUE)

layout <- "
AB
CD"

plot <- p1 + p2 + p3 + p4 + plot_annotation(tag_levels = 'A') + plot_layout(design=layout)

plot

ggsave(plot, filename = "../0-data/figures/Fig_2.pdf",
        width = 11,
        height = 7, dpi = 300)

```

\subsection{Figure 2. WID-HPV in relation to CIN, HPV oncogenic subtype and replicative age.}

**A** Overlayed dot-, box- and violin plots showing WID-HPV in the groups of different samples. Samples used for index training were excluded. Box plots show median and interquartile range, with whiskers showing 1.5 times the 25th and 75th percentile and dots the outliers. Violin plots give the kernel probability density of the data in each group. P value is derived from Wilcoxon test comparing all HPV+ cytology negative (cyt-) samples, CIN1, and CIN2 samples versus CIN3+. P values for difference to HPV- cyt- samples (reference) were as follows: HPV+, p=2-9; CIN1, p=1.7-19; CIN2, p=3.1-11; CIN3, p=1 (not shown in the figure). P values were adjusted for multiple comparisons using the Bonferroni method. **B** Proportion of oncogenic HPV subtypes detected by qPCR for each sample group defined in the CIN dataset. Samples used for training the WID-HPV index were excluded. Other oncogenic HPV subtypes detected were 31, 33, 35, 39, 45, 51, 52, 56, 58, 59, 66, 68, or a combination thereof. **C** Performance of the WID-HPV in HPV16/18 positive samples versus other oncogenic HPV genotypes in the CIN dataset. Samples used for index training and those where no HPV was detected were excluded. Box plots give median and interquartile range, with whiskers showing 1.5 times the 25th and 75th percentile and dots the outliers. **D** Kendall (œÑ) and Spearman (œÅ) correlations of the WID-HPV-index with the logarithm of the pcgtAge score for the subset of HPV+ women with respect to CIN.

\pagebreak

```{r, warning = FALSE, message=FALSE, fig.width=10,fig.height=8}
# apoptosis hallmark genes
# ref <- dat_cin_notrain$type %in% c("Control_POS", "CIN1", "CIN2")
# comp <- dat_cin_notrain$type == "CIN3+"
# 
# cin3_all <- signif(wilcox.test(dat_cin_notrain[ref,]$median_methylation_apoptosis,
#                                dat_cin_notrain[comp,]$median_methylation_apoptosis)$p.value,2)
# cin3_all <- sub("e","%.% 10^",cin3_all)#fix superscript scientific notation
# 
# medapop <- dat_cin_notrain %>%
#   ggplot(aes(x = type,
#              y = median_methylation_apoptosis, fill=type)) +
#   geom_half_boxplot(width=0.2)+
#   geom_half_violin() +
#   geom_half_point(transformation = position_jitter(), nudge=0.05, size=0.3,aes(color=type)) +
#   ylab("Median methylation\n(apoptosis gene promoters)") +
#   xlab("") +
#   scale_x_discrete(labels = type.labs) +
#   scale_fill_manual(name="", labels=type.labs, values=cin.colours2)  +
#   theme_classic() +
#   scale_color_manual(name="", labels=type.labs, values=cin.colours2) +
#   theme(axis.text.x=element_blank(),
#         axis.ticks.x=element_blank())  +
#   annotate("segment",
#            x = c(2, 4, 2, 3, 3, 5),
#            xend = c(2, 4, 4, 3, 5, 5),
#            y = c(0.036, 0.036, 0.038, 0.038, 0.045, 0.045),
#            yend = c(0.038, 0.038, 0.038, 0.045, 0.045, 0.044),
#            size = 0.5) +
#   annotate("text",
#            x = 4,
#            y = 0.046,
#            label = paste0("p == ", cin3_all),
#            size = 3.1,
#            colour = "gray40",
#            parse=TRUE)
# 
# ggsave(medapop, filename = "../0-data/figures/fig3b.pdf",
#        width = 5.5,
#        height = 3.5, dpi = 300)
# 
# 
# # apoptosis index CIN data
# ref <- dat_cin_notrain$type %in% c("Control_POS", "CIN1", "CIN2")
# comp <- dat_cin_notrain$type == "CIN3+"
# cin3_all <- signif(wilcox.test(dat_cin_notrain[ref,]$index_apoptosis,
#                     dat_cin_notrain[comp,]$index_apoptosis)$p.value,2)
# cin3_all <- sub("e","%.% 10^",cin3_all)#fix superscript scientific notation
# 
# apopind <- dat_cin_notrain %>%
#   ggplot(aes(x = type,
#              y = index_apoptosis, fill=type)) +
#   geom_half_boxplot(width=0.2)+
#   geom_half_violin() +
#   geom_half_point(transformation = position_jitter(), nudge=0.05, size=0.3,aes(color=type)) +
#   ylab("Apoptosis index") +
#   xlab("") +
#   scale_x_discrete(labels = type.labs) +
#   scale_fill_manual(name="", labels=type.labs, values=cin.colours2) +
#   scale_color_manual(name="", labels=type.labs, values=cin.colours2) +
#   theme_classic() +
#   theme(axis.text.x=element_blank(),
#         axis.ticks.x=element_blank()) +
#   annotate("segment",
#            x = c(2, 4, 2, 3, 3, 5),
#            xend = c(2, 4, 4, 3, 5, 5),
#            y = c(-6.25, -6, -5.5, -5.5, -5.25, -5.25),
#            yend = c(-5.5, -5.5, -5.5, -5.25, -5.25, -5.5),
#            size = 0.5) +
#   annotate("text",
#            x = 4,
#            y = -4.75,
#            label = paste0("p == ", cin3_all),
#            size = 3.1,
#            colour = "gray40",
#            parse = TRUE)
# ggsave(apopind, filename = "../0-data/figures/fig3c.pdf",
#        width = 5.5,
#        height = 3.5, dpi = 300)
# 
# 
# 
# # correlation WID-HPV and apoptosis index
# 
##first save correlation coefficients and p-values so can format them as wanted
#dat3 <- dat_cin_notrain %>%
#  filter(type %in% c("Control_POS","CIN1","CIN2","CIN3+")) %>%
#  droplevels()
#
#tauCor <- round(cor(dat3$index_apoptosis,dat3$WID_HPV_adj, method="kendall"),2)
#tauPval <- sub("e","%.% 10^",signif(cor.test(dat3$index_apoptosis,dat3$WID_HPV_adj, method="kendall")$p.value,2))
#rhoCor <- round(cor(dat3$index_apoptosis,dat3$WID_HPV_adj, method="spearman"),2)
#rhoPval <- sub("e","%.% 10^",signif(cor.test(dat3$index_apoptosis,dat3$WID_HPV_adj, method="spearman")$p.value,2))
#
#apopcorr <- dat3 %>%
#   ggplot(aes(x = index_apoptosis,
#              y = WID_HPV_adj)) +
#   geom_point(aes(colour=type), size=0.6) +
#   scale_color_manual(name="",labels=type.labs.3,values=cin.colours3) +
#   theme_classic() +
#   labs(y = "WID-HPV", x = "Apoptosis index", colour="") +
#  annotate("text",
#           x = -15,
#           y = 2.5,
#           label = paste0(expression(tau), "==", tauCor, "~(", "p == ", tauPval, ")"),
#           size = 3.1,
#           colour = "gray40",
#           parse = TRUE) +
#  annotate("text",
#           x = -15,
#           y = 2,
#           label = paste0(expression(rho), "==", rhoCor, "~(", "p == ", rhoPval, ")"),
#           size = 3.1,
#           colour = "gray40",
#           parse = TRUE)
#
#
#ggsave(apopcorr, filename = "../0-data/figures/fig3d.pdf",
#        width = 5.5,
#        height = 3.5, dpi = 300)

```

\includegraphics[trim=0 0cm 0 0, clip]{../0-data/figures/Fig_3.pdf}


\subsection{Figure 3. WID-HPV correlation with apoptosis.}

**A** Overview methods for evaluating apoptosis. **B** Overlayed dot-, box- and violin plots showing median methylation of CpGs in promotor regions from the Apoptosis Hallmark gene set (MSigDB) versus HPV and cytology status in the CIN dataset. P value derived from Wilcoxon test comparing HPV+, CIN1, and CIN2 with CIN3+ samples. **C** Overlayed dot-, box- and violin plots showing the apoptosis index constructed using BH3 mimetics on cancer cell lines versus HPV and cytology status in the CIN dataset. P value derived from Wilcoxon test comparing HPV+, CIN1, and CIN2 with CIN3+ samples. **D** Kendall (œÑ) and Spearman (œÅ) correlations of the WID-HPV-index with the apoptosis-index for the subset of women that were HPV+ or had CIN1-3+. Box plots give median and interquartile range, with whiskers showing 1.5 times the 25th and 75th percentile and dots the outliers. Violin plots give the kernel probability density of the data in each group. Samples used for training the WID-HPV were excluded from all plots.

\pagebreak

\subsection{Table 1. Annotation of the 27 WID-HPV-index CpGs.} CpGs were annotated using the IlluminaHumanMethylationEPICanno.ilm10b4.hg19 R package (Bioconductor). db_epithelial is the mean delta-beta of the epithelial component (inferred immune cell composition = 0) in HPV- compared to HPV+ cytology negative samples in the training set. The mean delta-beta is positive when the repective CpG is on average hypermethylated in the HPV+ group, negative when on average hypomethylated.
```{r, echo=FALSE, message = FALSE, warning = FALSE}

# commented out, embedded resulted PDF for formatting purposes

# load summary on index CpGs

#dat <- readRDS("../1-source-codes/analysis-pipeline/6-output/annotation_WID-HPV_index_CpGs.Rds")
#dat <- dat %>% dplyr::rename(Infinium.type=Infinium.Probe.Type) %>% select(!methylation.epithelial)
#colnames(dat) <- gsub("\\.", " ", colnames(dat))
#rownames(dat) <- NULL
#t1 <- kable(dat, align=rep('c', 7), "html") %>% kable_paper() %>% column_spec(3,italic=T)
#kableExtra::save_kable(t1, file = "../0-data/tables/table-main1.html")

#t1 <- dat %>%
#  gt() %>%
#  tab_style(
#    style = list(
#      cell_text(style = "italic")
#      ),
#    locations = cells_body(
#      columns = Gene
#    )
#  ) %>%
#  tab_style(
#    style = list(
#      cell_text(weight = "bold")
#      ),
#    locations = cells_column_labels()
#  ) %>%
#   tab_options(
#   table.font.size = 10,
#   table.font.names = "Arial",
#   table.width = px(650)
#   )
#t1 %>% gtsave("../0-data/tables/table-main1.html")

#pagedown::chrome_print("../0-data/tables/table-main1.html", output = "../0-data/tables/table-main1.pdf")

```
\includegraphics[trim=2cm 10cm 2cm 0cm, clip]{../0-data/tables/table-main1.pdf}
