---
title: "HPV-induced host epigenetic reprogramming is lost upon progression to high-grade cervical intraepithelial neoplasia"
author: "Supplementary Materials 1"
date: ""
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
header-includes:
  \usepackage{fontspec}
  \usepackage[utf8]{inputenc}
  \usepackage{pdfpages}
  \usepackage{graphicx}
  \setmainfont{Arial}
---

```{r, echo=FALSE, message = FALSE, warning = FALSE}
library(knitr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(kableExtra)
library(ggbeeswarm)
library(gghalves)
library(patchwork)
library(ggpubr)
library(openxlsx)
library(viridis)
library(ComplexHeatmap)
library(ggplotify)

# load helper functions
source("../1-source-codes/plot_roc_dichot.R")
source("../1-source-codes/GOplot_methyl.R")

knitr::opts_chunk$set(echo=FALSE, message = FALSE, warning = FALSE)

theme_set(theme_minimal())
```

Chiara Herzog\*, Charlotte D. Vavourakis\*, James E. Barrett, Gerlinde Karbon, Andreas Villunger, Jiangrong Wang, Karin Sundstr√∂m, Joakim Dillner, Martin Widschwendter


\* contributed equally

\tableofcontents

\subsection{Supplementary Table 1. Participant characteristics.}

```{r}
## code commented out - saved as PDF for formatting. code below describes step for generating tables
# library(dplyr)
# library(tidyverse)
# library(gtsummary)
# library(gt)
# theme_gtsummary_compact()
# 
# dat_cin <- readRDS("../0-data/pheno_P1.Rds")
# 
# dat_cin$set <- dat_cin$split
# 
# data <- dat_cin %>%
#   select(-c(grep("hepidish", colnames(dat_cin)), EGA_pheno_ID, pcgtage, ic, WID_HPV, median_methylation_apoptosis, index_apoptosis, split)) %>%
#   mutate(set = case_when(set == "training set" ~ "Training set",
#                          set == "hold out test set" ~ "Test set",
#                          set == "CIN validation set" ~ "CIN set")) %>%
#   mutate(set = factor(set, levels = c("Training set",
#                                       "Test set",
#                                       "CIN set"))) %>%
#   mutate(type = case_when(type %in% c("Control_NEG", "Control_POS") ~ "No cytological changes",
#                           type == "CIN1" ~ "CIN1",
#                           type == "CIN2" ~ "CIN2",
#                           type == "CIN3+" ~ "CIN3+"),
#          HPV = case_when(HPV == "NEG" ~ "HPV-",
#                          HPV == "POS" ~ "HPV+",
#                          HPV == "Unknown" ~ "Unknown",
#                          type == "CIN3+" & is.na(HPV) ~ "HPV+")) %>%
#   mutate(HPV_genotype2 = case_when(grepl("16", HPV_genotype) ~ "HPV16/18",
#                                    grepl("18", HPV_genotype) ~ "HPV16/18",
#                                   HPV_genotype=="no HPV detected" ~ "no HPV detected",
#                                   HPV_genotype == "other HR" ~ "other oncogenic HPV",
#                                   is.na(HPV_genotype) ~ "Unknown")) %>%
#   mutate(HPV_genotype2 = factor(HPV_genotype2, levels = c("HPV16/18",
#                                                         "other oncogenic HPV",
#                                                         "no HPV detected",
#                                                         "Unknown"))) %>%
#   mutate(type = factor(type, levels = c("No cytological changes",
#                        "CIN1",
#                        "CIN2",
#                        "CIN3+"))) %>%
#   mutate(age_group = if_else(age < 30, "below 30", "above 30")) %>%
#   select(set, HPV, age, age_group,type, HPV_genotype2) #make sure age_group is after age
# 
# 
# tbl <- data %>%
#   tbl_strata(strata = set,
#            .tbl_fun=
#              ~ .x %>%
#              tbl_summary(by = HPV,
#                          type = list(age ~ "continuous"),
#                          label = list(age ~ "Age (median, IQR)",
#                                       age_group ~ "Age group",
#                                       type ~ "Cytological diagnosis",
#                                       HPV_genotype2 ~ "HPV subtype")) %>%
#   modify_header(update = all_stat_cols() ~"**{level}**<br>n = {n}") %>%
#   bold_labels() %>%
#   modify_footnote(everything() ~ NA))
# 
# t1 <- tbl %>%
#   as_gt() %>%
#   text_transform(locations = cells_body(),
#                fn = function(x){
#                  paste0(ifelse(x == "0 (0%)", "", x))}
# ) %>%
#   tab_options(
#   table.font.size = 10,
#   table.font.names = "Arial",
#   table.width = px(650)
# )
# 
# t1 %>%
#   gtsave("../0-data/tables/table-s1.html")
# 
# pagedown::chrome_print("../0-data/tables/table-s1.html", output = "../0-data/tables/table-s1.pdf")
```

\includegraphics[trim=2cm 10cm 2cm 0cm, clip]{../0-data/tables/table-s1.pdf}


\pagebreak

\subsection{Supplementary Table 2. Pathways enriched by the 27 WID-HPV-index CpGs.} Table displays the GO terms within the category Biological Process with more than two significant genes (p-value < 0.05) containing an index CpG. Intermediate terms between the highest and lowest possible level along the combined terms' path of a reduced DAG were removed with the GOxploreR package. Level gives the hierarchy level in the DAG structure, N gives the number of genes in each term.

```{r, echo=FALSE, message = FALSE, warning = FALSE}

# commented out, embedded resulted PDF for formatting purposes

# load gene ontology summary on index CpG enriched terms
#dat <- readRDS("../1-source-codes/analysis-pipeline/8-output/summaryGOindex.Rds")
#dat$pval <- round(dat$pval, digits=4)
#dat <- dat %>% dplyr::rename(p.value=pval) 
#colnames(dat) <- gsub("\\.", "-", colnames(dat))
#rownames(dat) <- NULL
#kable(dat,"latex") %>% kableExtra::kable_styling(latex_options = c("HOLD_position","repeat_header","scale_down")) %>% column_spec(6,italic=T)

#t2 <- dat %>%
#  gt() %>%
#  tab_style(
#    style = list(
#      cell_text(style = "italic")
#      ),
#    locations = cells_body(
#      columns = Genes
#    )
#  ) %>%
#  tab_style(
#    style = list(
#      cell_text(weight = "bold")
#      ),
#    locations = cells_column_labels()
#  ) %>%
#   tab_options(
#   table.font.size = 10,
#   table.font.names = "Arial",
#   table.width = px(650)
#   )
#t2 %>% gtsave("../0-data/tables/table-s2.html")

#pagedown::chrome_print("../0-data/tables/table-s2.html", output = "../0-data/tables/table-s2.pdf")

```

\includegraphics[trim=2cm 10cm 2cm 0cm, clip]{../0-data/tables/table-s2.pdf}


```{r, echo=FALSE, include=FALSE, message = FALSE, warning = FALSE}

#commented out. Provided as separate Excel file

#Supplementary Table 3. Annotation of the 114 apoptosis index CpGs. CpGs were annotated using IlluminaHumanMethylationEPICanno.ilm10b4.hg19 R package (Bioconductor). Methylation status relates to mean delta-beta for control cell lines versus cell lines grown in the presence of BH3 mimetics in the training set.

## load summary on index CpGs
#dat <- readRDS("../1-source-codes/analysis-pipeline/14-output/annotation_apoptosis_index_CpGs.Rds")
#dat <- dat %>% dplyr::rename(Probe.type=Infinium.Probe.Type, Methylation.status=methylation.mean) 
#colnames(dat) <- gsub("\\.", " ", colnames(dat))
#wb <- createWorkbook()
#addWorksheet(wb, "Anno 114 apoptosis index Cpgs")
#writeDataTable(wb = wb, sheet = 1, x = dat, rowNames=FALSE) 
#saveWorkbook(wb, "./Supplementary_Table_4.xlsx")

```

\pagebreak

```{r, warning = FALSE, message=FALSE, fig.width=15,fig.height=15}

# data for this plot was generated in analysis-pipeline/8-output

# load and arrange GSEA data for bubble plots
table <- readRDS("../0-data/GSEA_training_WID_HPV/DMPs_vs_all_result_GO.Rds")
table <- table %>% filter(FDR<0.05) %>% rownames_to_column(var="ID") %>% droplevels # keep only significant pathways

# number of DMPs in all genes
gene_count <- readRDS("../0-data/GSEA_training_WID_HPV/DMP_count_siggenes_GO.Rds")

#format dataframes for plotting
gene_count <- gene_count %>% rownames_to_column(var="ID")
gene_count$DMP_count <- as.numeric(gene_count$DMP_count)
table <- table %>% 
  dplyr::rename(Category=ONTOLOGY,Term=TERM,Genes=SigGenesInSet, pval=P.DE, adj_pval=FDR)
circ <- circle_dat(table, gene_count)

# add number of DMPs per term to circ
count2 <- c()#init
terms <- unique(circ$ID)
for (i in 1:length(terms)){
  co <- sum(circ$DMP_count[circ$ID==terms[i]])
  count2 <- c(count2,co)
}
df <- data.frame(ID=terms, count2=count2)
circ <- full_join(circ,df)#combine data

# make a bubble plot
circ$count <- as.numeric(circ$count)
circ$pval_log <- -log(circ$pval,10)
circ$adj_pval_log <- -log(circ$adj_pval, 10)
circ$term <- paste0(circ$term," (", circ$ID,")")

# subplot go- BB
p1 <- circ %>%
  filter(category=="BP") %>% droplevels()%>%
  arrange(desc(adj_pval_log)) %>%
  ggplot(aes(x=count, y=count2, size=adj_pval_log, fill=term)) +
    geom_point(alpha=0.5, shape=21, color="#7A0177") +
    scale_size(range = c(min(circ$adj_pval_log), 2*max(circ$adj_pval_log))) +
  scale_fill_brewer(palette = "RdPu", direction=-1) +
  guides(size=guide_legend(ncol=3))+
  theme_minimal()+
  labs(size="-log(adj. p-val)", fill="") +
  xlab("Number of genes in term")+
  ylab("Number of DMPs in term")+
  ggtitle('Enriched GO terms, Biological process')

# subplot go - CC
p2 <- circ %>%
  filter(category=="CC") %>% droplevels()%>%
  arrange(desc(adj_pval_log)) %>%
  ggplot(aes(x=count, y=count2, size=adj_pval_log, fill=term)) +
    geom_point(alpha=0.5, shape=21, color="#005A32") +
    scale_size(range = c(min(circ$adj_pval_log), 2*max(circ$adj_pval_log))) +
  scale_fill_brewer(palette = "Greens", direction=-1) +
  guides(size=guide_legend(ncol=3))+
  theme_minimal()+
  labs(size="-log(adj. p-val)", fill="") +
  xlab("Number of genes in term")+
  ylab("Number of DMPs in term")+
  ggtitle('Enriched GO terms, Cellular component')
# note that membrane-enclosed lumen (GO:0031974),organelle lumen (GO:0043233),intracellular organelle lumen (GO:0070013) completely overlap

# subplot chord
table_BP <- table %>% filter(Category=="BP") %>% droplevels()
genes_BP <- c() #init
for (i in 1:length(table_BP$Genes)){
  out <- unlist(strsplit(table_BP$Genes[i],","))
  genes_BP <- c(genes_BP,out)
}
genes_BP <- unique(genes_BP)
gene_count_BP <- gene_count %>% filter(ID %in% genes_BP)
circ_BP <- circle_dat(table_BP, gene_count_BP)

# make matrix for chord plot
N <- length(table_BP$Term) + 1
M <- gene_count_BP[,1] 
mat <- matrix(0, ncol = N, nrow = length(M))
rownames(mat) <- M
colnames(mat) <- c(table_BP$ID, 'DMPcount') 
mat[,N] <- gene_count_BP[,2]
for (p in 1:(N-1)){
  dat <- subset(circ_BP, ID == table_BP$ID[p])
  for (g in 1:length(M)) mat[g, p] <- ifelse(M[g]%in%dat$genes, 1, 0)
    }
chord <- mat

p3 <- GOChord(chord,space = 0.02, limit= c(7,0) , gene.order = 'DMPcount', gene.space = 0.25, gene.size=3) # displaying only genes in minimum 7 categories, note that this function still doesn't work for setting limits higher on DMPcount :/
p3 <- p3 + ggtitle("Genes encompassing minimum 7 enriched GO terms, Biological process")


# subplot heatmap
## read in pre-calculated delta-beta values
db <- readRDS("../0-data/GSEA_training_WID_HPV/deltabeta_train_dmp.Rds")
annot_DMP_mM <- readRDS("../0-data/GSEA_training_WID_HPV/DMPs_annot_missMethyl.Rds")
cpg_8386_annot <- readRDS("../0-data/GSEA_training_WID_HPV/cpg_8386_annot.Rds") #need this file for CpG context information


## filter out genes and CpGs from GO term of interest
genes_GO_interest <- unlist(strsplit(table$Genes[table$Term=="positive regulation of DNA repair"],","))# 43 genes in term
annot_CpGs_GO_interest <- annot_DMP_mM %>% filter(alias %in% genes_GO_interest) #62 CpGs in 43 genes

## start with arranging factor gene, cause this determines order CpGs in final plot
gene <- annot_CpGs_GO_interest %>% dplyr::select(cpg,alias)
gene <- gene[with(gene, order(alias)),]
gene$alias <- factor(gene$alias, levels=c(unique(gene$alias)))
order_of_plot <- c(gene$cpg)
gene <- gene$alias

## make a matrix of the delta-beta values training set (main heatmap)
mat1 <- db %>% dplyr::select(c(db_epithelial,db_immune,db_mean)) %>% filter(row.names(db) %in% unique(annot_CpGs_GO_interest$cpg))
mat1 <- mat1[match(order_of_plot, row.names(mat1)),] #match order CpGs
#identical(row.names(mat1), order_of_plot) #sanity-check

## factor cgi 
cgi <- cpg_8386_annot %>% filter(Name %in% row.names(mat1)) %>% dplyr::select(Name,Relation_to_Island) %>% droplevels()
cgi$Relation_to_Island <- factor(cgi$Relation_to_Island, levels=c("N_Shelf","N_Shore","Island","S_Shore","S_Shelf","OpenSea"))
cgi <- cgi[match(order_of_plot, cgi$Name),] #match order CpGs
#identical(order_of_plot, as.character(cgi$Name)) #sanity-check
cgi <- cgi$Relation_to_Island

### factor GO term - could also add the occurence of these genes in other GO terms

### colors
cgi_col = c("Island" = magma(20)[5] ,"OpenSea" = magma(20)[18], "S_Shelf" = magma(20)[13],  "S_Shore" = magma(20)[9], "N_Shelf" = magma(20)[14],  "N_Shore" = magma(20)[10])
#col_fun = colorRamp2(c(-0.05, 0, 0.15), c("darkblue", "white", "darkred"))

### heatmap
ht_list =
  Heatmap(mat1, col = viridis(100), name = "Delta-beta", row_names_side = "left", row_names_gp = gpar(fontsize = 8), cluster_rows = FALSE, show_column_dend = FALSE, column_names_rot = 60, column_names_centered = TRUE,row_split = c(gene), row_title_rot = 0, row_title_gp = gpar(fontsize=8), cluster_row_slices=FALSE, border=TRUE, column_title = "Genes in term \n Positive Regulation of DNA repair") + 
  Heatmap(cgi, name = "CGI",  width = unit(5, "mm"), cluster_rows=FALSE,
    col = cgi_col, column_names_rot = 0, column_names_centered = TRUE)

p4 <- grid2grob(draw(ht_list, merge_legends = TRUE, heatmap_legend_side = "right"))

# print full plot
layout <- "
AAADD
BBBDD
CCCDD
CCCDD
"

plot <- p1 + p2 +p3 + p4 + plot_annotation(tag_levels = 'a') + plot_layout(design = layout)

plot


```

\subsection{Supplementary Figure 1. Differential methylation and gene ontology analysis in HPV-infected and -uninfected samples.}

\pagebreak

```{r, warning = FALSE, message=FALSE, fig.width=15,fig.height=12}

cell_cols <- c("#7A12FF","#1E82FF","#10D0E8","#1CFFB0")
apo_cols <- c("#D95F02","#109DB7","#D8C99E")
OC_cols <- c("#FF5C0D","#FF008A","#BC0CE8","#5B03FF")


## percentage of dead cells, controls vs mims at different time points
df <- readRDS("../0-data/Cell_line_test_BH3mims.Rds")

p1 <- df %>%
  ggplot(aes(x=timepoint, 
             y=DC_percent, fill = cell_line)) +
  geom_bar(position="dodge", stat="identity")+
  geom_hline(yintercept=50, linetype="dashed", color = "black") +
  facet_wrap(~cell_line, strip.position = "top",scales = "free_x", ncol=4) +
  theme_minimal() +
  theme(panel.spacing.x = unit(0, "pt"), 
        strip.placement = "outside", 
        strip.background.x = element_blank(),
        axis.line.x = element_line(size = .1),
        panel.grid.major.y = element_line(linetype = "dotted"),
        panel.grid.major.x = element_blank(),
        panel.grid.minor = element_blank(),
        legend.position="") +
  xlab("Timepoint/treatment") +
  ylab("% dead cells") +
  #ylim(0,100)+
  labs(fill="") +
  scale_fill_manual(values=cell_cols) +
  annotate("rect", xmin = 0.5, xmax=2.5, ymin = -5, ymax=-2.5, fill = apo_cols[1]) +
  annotate("rect", xmin =2.5, xmax=3.5, ymin = -5, ymax=-2.5, fill = apo_cols[2]) +
  annotate("rect", xmin=3.5,xmax=4.5, ymin = -5, ymax=-2.5, fill = apo_cols[3]) +
  ggtitle("Induction of apoptosis with BH3 mimetics")

#make dataframe for annotation label
#dat_text <- data.frame(cell_line = levels(df$cell_line),
                       #labels = c("","","- = DMSO (control) \n + = S63845/ABT-737",""))

#p1 <- p1 + geom_text(data = dat_text, mapping = aes(x = 2, y = 90, label = labels, hjust=0)) 

# Draw ggplot2 plot with custom legend only
legend <- ggplot() + xlim(0, 1) + ylim(0, 1) +                  
  annotate("text",x = 0,y = 1,size = 5,label = " - = DMSO (control) \n + = S63845/ABT-737", hjust=0, vjust =1) +theme_void()

## performance apoptosis index in train and test set
dat <- readRDS("../1-source-codes/analysis-pipeline/15-output/pheno_mims_apo_index.Rds")

sets.labs <- c(paste("Training set \n n=", sum(dat$split=='training set')), 
               paste("Test set \n n=", sum(dat$split=='hold out test set')))
names(sets.labs) <- c("training set", "hold out test set")

dat$BH3.mims_type <- factor(dat$BH3.mims_type, levels=c("control", "early apoptosis", "late apoptosis"))

p2 <- dat %>%
  ggplot(aes(x = split,
             y = apop_index, color = BH3.mims_type)) +
  geom_jitter(width=0.25) +
  labs(y = "Apoptosis index", x = "") +
  scale_x_discrete(labels = sets.labs) +
  geom_hline(yintercept=0, linetype="dashed", color = "black") +
  scale_color_manual(values = apo_cols)  +
  ylim(-1.5,1.5)+
  labs(color="") +
  ggtitle("Index performance: BH3 mimetics")

#legend2 <- get_legend(p2) 

## performance apoptosis index in HGSOC external set
dat <- readRDS("../1-source-codes/analysis-pipeline/15-output/pheno_GSE168225.Rds")

#rearrange factors
dat$type <- factor(dat$type, levels=c("control","azacitidine+carboplatin"))
dat$characteristics_ch1.1 <- gsub("cell line: ", "", dat$characteristics_ch1.1)

p3 <- dat %>%
  ggplot(aes(x = type, 
             y= apop_index, group=characteristics_ch1.1)) + 
  geom_point() +
  geom_line() +
  labs(y = "Apoptosis index", x = "")+
  aes(color=characteristics_ch1.1) +
  scale_color_manual(values=OC_cols)+
  labs(color="") +
  ggtitle("External validation: HGSOC cell lines")

# final plot

legend <- legend + plot_layout(tag_level = 'new') #avoid labeling legend

layout <- 
"
AAAD
AAA#
BBCC
BBCC"

p1 + p2 +p3 + plot_annotation(tag_levels = 'a') + legend + plot_layout(design=layout)

```

\subsection{Supplementary Figure 2. Experimental design of BH3 mimetics treatment and development of the apoptosis index.}

**A** Four cancer cell lines were grown in the presence of the BH3 mimetics S63845 and ABT-737 and collected at two different time points, reflecting early and late stages of apoptosis, respectively. "Control" cancer cell lines were grown in the presence of DMSO and pooled for the two time points, as no significant apoptosis was expected at either timepoints when incubating only in DMSO. **B** Apoptosis index performance in the test set and training set. **C** Apoptosis index calculated for an EPIC methylation array dataset of high-grade serous ovarian cancer cell lines in which apoptosis was induced with azacitidine and carboplatin treatment. 10-30% apoptosis were reported by the authors of the original study (Wong-Brown et al., 2022). 

\pagebreak

\includegraphics[trim=0 0cm 0 0, clip]{../0-data/figures/Fig_S3.pdf}

\subsection{Supplementary Figure 3. Overview of cervical screening samples used in this study.}